### PR TITLE
Circuit breaker backport

### PR DESCRIPTION
**Pull request by rachedbenmustapha**

### CODE REVIEW

File: extensions/lifecycle/conductor/LifecycleConductor.js [hunk 1]
```diff
@@ -427,7 +492,19 @@ class LifecycleConductor {
                 }, next);
             },
             () => isTruncated,
-            err => cb(err, nEnqueued));
+            err => {
+                if (!err) {
```
Comment (jonathan-gramain) [inline extensions/lifecycle/conductor/LifecycleConductor.js]: Shouldn't we also call `cb` when there is an error?

File: extensions/lifecycle/conductor/LifecycleConductor.js [hunk 2]
```diff
@@ -542,6 +619,12 @@ class LifecycleConductor {
         });
     }
 
+    needsZookeeper() {
+        return this._bucketSource === 'zookeeper' || // bucket list stored in zk
+            this._bucketSource === 'mongodb' || // bucket stream checkpoints in zk
+            this._bucketSource === 'bucketd'; // bucket stream checkpoints in zk
```
Comment (jonathan-gramain) [inline extensions/lifecycle/conductor/LifecycleConductor.js:627]: Little more compact (feel free to ignore):

```suggestion
        return ['zookeeper', 'mongodb', 'bucketd'].includes(this._bucketSource);
```

File: extensions/lifecycle/conductor/LifecycleConductor.js (hunk 1 above)
Comment (rachedbenmustapha) [reply extensions/lifecycle/conductor/LifecycleConductor.js]: You're right, seems like the cb call was wrongly deduped.

**Review by nicolas2bert — APPROVED** [review]
LGTM, once the callback remark is addressed.

**Review by tmacro — APPROVED** [review]

**Review by jonathan-gramain — APPROVED** [review]

**Pull request merged**